File: Pyhton3 Implementations/Searching/Array/LinearSearch.py
from typing import List
def Linear_Search(nums : List[int], target : int) -> int:
    """
    Does a Linear search on the list and returns the index position of 
    the target if found else returns -1

    input:
        nums -> List of numbers of type integer
        target -> The value to be searched for in the list nums
    Output:
        index -> The index position in which the target is found
                If not found returns -1
    """
    
    for ind, val in enumerate(nums):
        if val == target:
            return ind
        
    return -1

File: JAVA Implementations/src/Searching/LinearSearch/LinearSearch.java
package Searching.LinearSearch;

public class LinearSearch {
    public static void main(String[] args) {
        int[] array = {1,2,5,67,89,21};
        int ans = search(array,67);
        if(ans == -1){
            System.out.println("Element not Found");
        }else{
            System.out.println("Element found at index:" + ans);
        }
    }
    public static int search(int[] array,int target){
        for(int i=0;i<array.length;i++){
            if(array[i] == target){
                return i;
            }
        }
        return -1;
    }
}
